### PR TITLE
Fix ability to edit pages via GitHub edit link

### PR DESCRIPTION
**Pull request by tpdorsey**
## Summary

Fix misconfigured edit path to enable submitting edits via the "Pencil" edit link.

In addition:

- Adds yaml frontmatter to doc pages
- Frontmatter includes a note indicating that API pages are generated from docstrings in source files. This gives some indication to a user who wants to edit the API docs, but selects the edit pencil and finds a basically blank page. 

## Checklist
- [X] Summarized PR's changes in the **Unreleased** section of the [CHANGELOG.md](https://github.com/PrefectHQ/prefect-aws/blob/main/CHANGELOG.md)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,12 @@
+---
+description: prefect-aws provides pre-built Prefect tasks for working with Amazon Web Services.
+tags:
+    - overview
+    - getting started
+    - installation
+    - examples
+---
```
Comment (desertaxle) [inline README.md:8]: This section gets rendered as a table by GitHub: 
Screen Shot 2022-09-16 at 10 13 03 AM

I think the best course of action would be to inject this section into this page into the `gen_refs` script that allows us to use README.md as the index page for our docs.

**Review by desertaxle — CHANGES_REQUESTED** [review]
Thanks @tpdorsey! I have one suggestion for the README, but other than that this looks great!

File: README.md (hunk 1 above)
Comment (tpdorsey) [reply README.md:8]: Ah I see my mistake here. We should remove this from the readme. My bad. It's fine on the other pages.

File: docs/batch.md [hunk 2]
```diff
@@ -1 +1,6 @@
+---
+description: Tasks for interacting with AWS Batch
+notes: This documentation pages is generated from source file docstrings.
```
Comment (desertaxle) [inline docs/batch.md:3]:
```suggestion
notes: This documentation page is generated from source file docstrings.
```

File: docs/client_waiter.md [hunk 3]
```diff
@@ -1 +1,6 @@
+---
+description: Task for waiting on a long-running AWS job
+notes: This documentation pages is generated from source file docstrings.
```
Comment (desertaxle) [inline docs/client_waiter.md:3]:
```suggestion
notes: This documentation page is generated from source file docstrings.
```

File: docs/credentials.md [hunk 4]
```diff
@@ -1 +1,6 @@
+---
+description: Module handling AWS credentials
+notes: This documentation pages is generated from source file docstrings.
```
Comment (desertaxle) [inline docs/credentials.md:3]:
```suggestion
notes: This documentation page is generated from source file docstrings.
```

File: docs/ecs.md [hunk 5]
```diff
@@ -1 +1,6 @@
+---
+description: Integrations with the Amazon Elastic Container Service.
+notes: This documentation pages is generated from source file docstrings.
```
Comment (desertaxle) [inline docs/ecs.md:3]:
```suggestion
notes: This documentation page is generated from source file docstrings.
```

File: docs/s3.md [hunk 6]
```diff
@@ -1 +1,6 @@
+---
+description: Tasks for interacting with AWS S3
+notes: This documentation pages is generated from source file docstrings.
```
Comment (desertaxle) [inline docs/s3.md:3]:
```suggestion
notes: This documentation page is generated from source file docstrings.
```

File: docs/secrets_manager.md [hunk 7]
```diff
@@ -1 +1,6 @@
+---
+description: Tasks for interacting with AWS Secrets Manager
+notes: This documentation pages is generated from source file docstrings.
```
Comment (desertaxle) [inline docs/secrets_manager.md:3]:
```suggestion
notes: This documentation page is generated from source file docstrings.
```

**Review by ahuang11 — APPROVED** [review]

**Review by desertaxle — APPROVED** [review]

**Pull request merged**